plugins/services/src/js/components/modals/ServiceRootGroupModal/index.tsx

```diff
@@ -0,0 +1,479 @@
+import FieldHelp from "#SRC/js/components/form/FieldHelp";
+import FluidGeminiScrollbar from "#SRC/js/components/FluidGeminiScrollbar";
+import FormRow from "#SRC/js/components/form/FormRow";
+import FormGroup from "#SRC/js/components/form/FormGroup";
+import FormGroupHeading from "#SRC/js/components/form/FormGroupHeading";
+import FormGroupHeadingContent from "#SRC/js/components/form/FormGroupHeadingContent";
+import FullScreenModal from "#SRC/js/components/modals/FullScreenModal";
+import FieldError from "#SRC/js/components/form/FieldError";
+import InfoTooltipIcon from "#SRC/js/components/form/InfoTooltipIcon";
+
+import { formatQuotaID } from "#PLUGINS/services/src/js/utils/QuotaUtil";
+import {
+  GroupFormData,
+  GroupFormErrors
+} from "#PLUGINS/services/src/js/types/GroupForm";
+
+import GroupModalHeader from "./Header";
+import ErrorsPanel from "./ErrorsPanel";
+import {
+  emptyGroupFormData,
+  getPathFromGroupId,
+  validateGroupFormData
+} from "./utils";
+
+const dl = container.get<DataLayer>(DataLayerType);
+const i18n = container.get<I18n>(TYPES.I18n);
+
+const groupCreateMutation = gql`
+  mutation {
+    createGroup(data: $data)
+  }
+`;
+
+function getSaveAction(data: GroupFormData) {
+  const newID = formatQuotaID(data.id);
+  const newData = (({ id, ...other }) => ({ id: newID, ...other }))(data);
+  return dl.query(groupCreateMutation, {
+    data: newData
+  });
+}
+
+interface ServiceRootGroupModalState {
+  isOpen: boolean;
+  isPending: boolean;
+  expandAdvancedSettings: boolean;
+  data: GroupFormData;
+  errors: GroupFormErrors;
+}
+
+interface ServiceRootGroupModalProps {
+  isEdit: boolean;
+}
+
+const METHODS_TO_BIND: string[] = [
+  "getAdvancedSettings",
+  "getModalContent",
+  "handleAdvancedSectionClick",
+  "handleClose",
+  "handleFormChange",
+  "handleSave"
+];
+
+class ServiceRootGroupModal extends React.Component<
+  ServiceRootGroupModalProps,
+  ServiceRootGroupModalState
+> {
+  static contextTypes = {
+    router: routerShape
+  };
+  static defaultProps = {
+    isEdit: false
+  };
+
+  constructor() {
+    // @ts-ignore
+    super(...arguments);
+
+    this.state = this.getInitialState();
+
+    METHODS_TO_BIND.forEach(method => {
+      // @ts-ignore
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  getInitialState(
+    _props: ServiceRootGroupModalProps = this.props
+  ): ServiceRootGroupModalState {
+    return {
+      isOpen: true,
+      isPending: false,
+      expandAdvancedSettings: false,
+      data: emptyGroupFormData(),
+      errors: {}
+    };
+  }
+
+  handleClose() {
+    // Start the animation of the modal by setting isOpen to false
+    this.setState(
+      { isOpen: false, isPending: false, data: emptyGroupFormData() },
+      () => {
+        // Once state is set, start a timer for the length of the animation and
+        // navigate away once the animation is over.
+        setTimeout(this.context.router.goBack, 300);
```

Comment:
This seems unorthodox and brittle with respect to animation timing. I can't think of something better that does not involve always rendering this modal.

Reply:
This came directly from the `CreateServiceModal`